src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/UpToDateCheckInput.xaml

```diff
@@ -1,16 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.-->
-<Rule
-    Name="UpToDateCheckInput"
-    DisplayName="Up-to-date check input"
-    PageTemplate="generic"
-    Description="File Properties"
-    xmlns="http://schemas.microsoft.com/build/2009/properties">
+<Rule Name="UpToDateCheckInput" xmlns="http://schemas.microsoft.com/build/2009/properties">
     <Rule.DataSource>
-        <DataSource Persistence="ProjectFile" HasConfigurationCondition="False" ItemType="UpToDateCheckInput" SourceOfDefaultValue="AfterContext" />
+        <DataSource Persistence="ProjectFile" HasConfigurationCondition="False" SourceOfDefaultValue="AfterContext" 
+                    SourceType="TargetResults" MSBuildTarget="CollectUpToDateCheckInputDesignTime"/>
     </Rule.DataSource>
-    <Rule.Categories>
-        <Category Name="Advanced" DisplayName="Advanced" />
-        <Category Name="Misc" DisplayName="Misc" />
-    </Rule.Categories>
+    <BoolProperty Name="Visible" Visible="False" ReadOnly="True" />
```

Comment:
Do we need this property?

Reply:
I thought it was needed to keep the items from showing up in the solution explorer, but I might be incorrect about that.